src/connectors/articleService.ts

```diff
@@ -1385,27 +1385,33 @@ export class ArticleService extends BaseService {
     }
 
     // logged-in user
-    // calculate heart beat lapsed time in secondes
+    // calculate heart beat lapsed time in milisecondes
     const lapse = Date.now() - new Date(oldData.updatedAt).getTime()
 
     // calculate last read total time
     const readLength = Date.now() - new Date(oldData.lastRead).getTime()
 
-    // if original read longer than 30 minutes
-    // skip
-    if (userId && readLength > MINUTE * 30) {
+    // if lapse shorter than 5 minutes or readLength longer than 30 minutes, 
+    // only update the updatedAt and then return
+   if (lapse < MINUTE * 5 || readLength > MINUTE * 30) {
+      await this.baseUpdate(
+          oldData.id,
+          {
+            readTime
```

Comment:
`readTime` is defined on [line 1416](https://github.com/thematters/matters-server/pull/2078/files#diff-6775d7dc5f0198959c6fe0e20d1aeff1620ba32b75bb328a09c7d042230da5c9R1416), so we need to move the definition before this line.